supplier_app/views.py

```diff
@@ -101,7 +101,24 @@ class CompanyListView(LoginRequiredMixin, ListView):
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        context['all_companies'] = Company.objects.values('name')
+
+        all_companies_user_emails = []
+
+        for company in self.get_queryset():
+            emails = CompanyUserPermission.objects\
+                .filter(company=company) \
+                .select_related('user') \
+                .values('user__email') \
+                .distinct()
+
+            all_companies_user_emails.append({
+                'name': company.name,
+                'description': company.description,
+                'users_email': '\n'.join([email['user__email'] for email in emails])
+            })
```

Comment:
Instead of self.get_queryset(), we could use the already executed query in context['object_list'].
Also I think we could save the emails in a list in every company instead of using all_companies_user_emails

Reply:
Done it!